src/agent/include/agent.hpp

```diff
@@ -36,30 +38,33 @@ class Agent
     void Run();
 
 private:
-    /// @brief Queue for storing messages
-    std::shared_ptr<MultiTypeQueue> m_messageQueue;
-
-    /// @brief Pointer to a custom ISignalHandler implementation
-    std::unique_ptr<ISignalHandler> m_signalHandler;
-
     /// @brief Task manager
     TaskManager m_taskManager;
 
+    /// @brief System info
+    SysInfo m_sysInfo;
```

Comment:
Previously you were just creating a unique pointer instead of having an instance. If we can still do that we could move the include to the source file.